File: plusOne/plusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        if(digits[digits.length-1]==9){
            int i = digits.length;
            StringBuilder str = new StringBuilder();
            while(i>0){
                i--;
                if(digits[i]==9){
                    digits[i]=0;
                    if(i==0){
                        int[] newDigits = new int[digits.length+1];
                        for(int j = newDigits.length-1;j>0;j--){
                            newDigits[j]=digits[j-1];
                        }
                        newDigits[0]=1;
                        digits = newDigits;
                    }
                }
                else{
                    digits[i]++;
                    break;
                }
            }
        }
        else{
            digits[digits.length-1]++;
        }
        return digits;
    }
}

File: plusOne/plusOne.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        if digits[-1]==9:
            i = len(digits)-1
            while(i>=0):
                if digits[i]==9:
                    digits[i]=0
                    if i==0:
                        digits.insert(0,1)
                else:
                    digits[i]+=1
                    break
                i-=1
        else:
            digits[-1]+=1
        return digits